src/Integration.Vsix/Suppression/DelegateInjector.cs

```diff
@@ -61,8 +71,37 @@ private void OnAssemblyLoad(object sender, AssemblyLoadEventArgs args)
 
         private void InjectSuppressionDelegate(Assembly asm)
         {
-            // TODO: if this is a Sonar analyzer assembly, try to set the suppression delegate
+            // If this is a Sonar analyzer assembly, try to set the suppression delegate
             // Note: the property might not exist for down-level versions of the analyzer
+            const string AssemblyName = "SonarAnalyzer";
+            const string NamespaceName = "SonarAnalyzer.Helpers";
+            const string ClassName = "SonarAnalysisContext";
+            const string PropertyName = "ShouldDiagnosticBeReported";
```

Comment:
I prefer `ShouldReportDiagnostic` a bit more, but this is a matter of taste and discussion :)

Reply:
I'm happy to discuss it :-) I used this name because it is follows the same form as the existing poor man's DI property _ShouldAnalysisBeDisabled_, but the name is quite clumsy.